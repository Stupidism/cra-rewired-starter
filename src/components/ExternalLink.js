import React from 'react';

export default ({ to, children }) => (
  <a href={to} target="_blank" rel="noopener noreferrer">
    {children}
  </a>
);
