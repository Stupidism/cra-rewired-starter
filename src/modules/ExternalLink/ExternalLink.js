import React from 'react';

const ExternalLink = ({ to, children }) => (
  <a href={to} target="_blank" rel="noopener noreferrer">
    {children}
  </a>
);

export default ExternalLink;
