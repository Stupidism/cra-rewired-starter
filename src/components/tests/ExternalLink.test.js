import React from 'react';
import { mount } from 'enzyme';

import ExternalLink from '../ExternalLink';

describe('<App />', () => {
  it('should be consistent with snapshot', () => {
    const snapshot = mount(<ExternalLink to="some-url" />);
    expect(snapshot).toMatchSnapshot();
  });

  it('should render an `a` tag element', () => {
    const wrapper = mount(<ExternalLink to="some-url">link</ExternalLink>);
    expect(wrapper.getDOMNode().innerHTML).toBe('link');
    expect(wrapper.getDOMNode().getAttribute('href')).toBe('some-url');
    expect(wrapper.getDOMNode().getAttribute('rel')).toBe(
      'noopener noreferrer',
    );
    expect(wrapper.getDOMNode().getAttribute('target')).toBe('_blank');
  });
});
