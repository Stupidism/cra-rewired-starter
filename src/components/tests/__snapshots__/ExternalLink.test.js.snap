// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should be consistent with snapshot 1`] = `
<a
  href="some-url"
  rel="noopener noreferrer"
  target="_blank"
/>
`;
